### PR TITLE
feat: implement `set_menu` for system tray

### PR DESCRIPTION
**Pull request by fnune**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by fnune** [comment]
I'm not sure if this works. I'm not sure how `_window_target` is being passed to these functions.

File: src/system_tray.rs [hunk 1]
```diff
@@ -113,4 +114,12 @@ impl SystemTray {
   pub fn set_icon(&mut self, icon: PathBuf) {
     self.0.set_icon(icon)
   }
+
+  pub fn set_menu<T: 'static>(
+    &mut self,
+    tray_menu: MenuPlatform,
```
Comment (amrbashir) [inline src/system_tray.rs]: tray_menu should be of type `ContextMenu` and not `MenuPlatform`

**Comment by fnune** [comment]
Oh sweet! You're extending it. I'll have time to work on this tonight, or maybe over the weekend

**Comment by lemarier** [comment]
Ok, so It works fine on `Linux` and `macOS`.

The only issue we have is `Windows`, once we set the menu the tray menu stops working.

Gonna try to dig in. @amrbashir you have an idea?

**Comment by amrbashir** [comment]
yeah I saw that issue yesterday, It was because the `Menu` gets dropped which calls `DestroyMenu()` and it destroys the `HMENU` stored in the `SystemTray` and the stash

**Comment by fnune** [comment]
You guys rock! I wasn't expecting you to just take over the PR :smile:

**Review by lemarier — APPROVED** [review]

**Review by amrbashir — COMMENTED** [review]
@lemarier Why disabling this for windows ?
I think it is fine that the menu is dropped, if they want to set a new menu, they probably need to keep it alive as possible either build it within the same lifetime of event_loop or just static lifetime.

**Comment by lemarier** [comment]
> @lemarier Why disabling this for windows ?
> I think it is fine that the menu is dropped, if they want to set a new menu, they probably need to keep it alive as possible either build it within the same lifetime of event_loop or just static lifetime.

I wonder if we shouldn't handle it internally? Because it's not how macOS and Linux work, but you are right... maybe we can ship it anyway and see how it goes

**Comment by amrbashir** [comment]
> I wonder if we shouldn't handle it internally? Because it's not how macOS and Linux work, but you are right... maybe we can ship it anyway and see how it goes

Well, we can either add drop impl for linux and macOS or remove all drop impls.

**Comment by lemarier** [comment]
> > I wonder if we shouldn't handle it internally? Because it's not how macOS and Linux work, but you are right... maybe we can ship it anyway and see how it goes
> 
> Well, we can either add drop impl for linux and macOS or remove all drop impls.

on windows did the menus are linked to the app? so when closed they are all dropped anyway?

**Comment by amrbashir** [comment]
> on windows did the menus are linked to the app? so when closed they are all dropped anyway?

I think so.

**Comment by lemarier** [comment]
Thanks for your contribution @fnune we do appreciate it a lot!

I'm sure the community will enjoy this new feature.

Have a great day!

**Pull request merged**